sdk/src/client/api/block_builder/mod.rs

```diff
@@ -34,6 +35,17 @@ impl ClientInner {
 
         let protocol_params = self.get_protocol_parameters().await?;
 
+        let mut basic_block_body = BlockBody::build_basic(issuance.strong_parents()?)
+            .with_weak_parents(issuance.weak_parents()?)
+            .with_shallow_like_parents(issuance.shallow_like_parents()?)
+            .with_payload(payload)
+            .finish()?;
+
+        basic_block_body.set_max_burned_mana_to_minimum(
```

Comment:
we could pass the params and the rmc to the `finish` method. I also tried that but found it very ugly :shrug:

Reply:
I don't think it's ugly, it's less code and doesn't break our non-mutability promise. It's very similar to what we did with StorageScore I guess.

Reply:
Having it like for the outputs with the minimum amount would probably be good
So two `new` methods, one where the user provides the amount and one where the user provides the WorkScoreParameters with the reference_mana_cost which then calculates the minimum?

Reply:
I agree, it should be the exact same thing

Reply:
aight

Reply:
I had it with another `new` function, but then we need optional fields in the builder just for them. It makes much more sense to pass them in the finish method to prevent this.

Reply:
dealt with using the `enum` approach